Error: invalid pattern: undefined

#### Description of the bug/issue
After upgraded from 0.16 to beta.1 I'm now getting an error message and my sagas bomb out and don't run.  Any ideas? 
#### Steps to reproduce the bug/issue

1. First saga runs via throttle and outputs action
Saga:
```
function * doStats(){
  if (moment().subtract(statIntervalSec,'seconds').valueOf() > lastTime.valueOf())
  {
    lastTime = moment();
    global.log.info('doing stats');
    yield deviceInfo();
    yield epilogInfo();
  }}

export function * doStatsSagaWithDelay(){
  yield throttle(statIntervalSec*1000,'*',doStats);
}
```
Action output
```
{type:"STAT_PHONE"
value:Object
batteryLevel:100
deviceId:"msm8998"
freeDiskStorage:"1677MB"
maxMemory:"263MB"
totalDiskStorage:"-1194MB"
totalMemory:"7904MB"
type:"STAT_PHONE"
usedMemory:"tbd"
Symbol(@@redux-saga/SAGA_ACTION):true

```
2. I see this message in console and saga's no longer run
```
ExceptionsManager.js:71 Error: invalid pattern: undefined
    at matcher (blob:file:///e9088396-ba97-4270-b1c3-e826e1c1584d:90118:13)
    at blob:file:///e9088396-ba97-4270-b1c3-e826e1c1584d:90077:16
    at Array.some (native)
    at Function.<anonymous> (blob:file:///e9088396-ba97-4270-b1c3-e826e1c1584d:90076:23)
    at put (blob:file:///e9088396-ba97-4270-b1c3-e826e1c1584d:89847:34)
    at Object.chan.put (blob:file:///e9088396-ba97-4270-b1c3-e826e1c1584d:89878:9)
    at blob:file:///e9088396-ba97-4270-b1c3-e826e1c1584d:89156:19
    at dispatch (blob:file:///e9088396-ba97-4270-b1c3-e826e1c1584d:71539:20)
    at blob:file:///e9088396-ba97-4270-b1c3-e826e1c1584d:89509:14
    at blob:file:///e9088396-ba97-4270-b1c3-e826e1c1584d:90705:54 ""

15:34:00.950 ExceptionsManager.js:71 The above error occurred in task deviceInfo
    created by rootSaga
Tasks cancelled due to error:
scanRequestHandler
deviceFoundHandler
scanStoppedHandler
disconnectedDeviceHandler
getRSSIforConnectedDevices
connectionSuccessHandler
firmwareLoadedHandler
loadSupportedFwVersion
deviceInfoLoadedHandler
stateRquesthandler
batteryAndImpRequestHandler
deleteRecordingsHandler
requestMissingPageAfterRT
handleMissingPagesAndEventsOnDock
readCharHandler
requestRecordingsHandler
doStatsSaga
doStatsSagaWithDelay
epilogConnectedSaga
crActionsOfInterest
storeCompletedEegPages 

```

package.json
```
...
  "react": "16.3.1",
    "react-addons-update": "^15.6.2",
    "react-mixin": "^4.0.0",
    "react-native": "0.55.4",
    "react-redux": "^5.0.7",
    "redux": "^3.7.2",
    "redux-saga": "^1.0.0-beta.1",
...
```

--- a/packages/core/src/internal/io.js
+++ b/packages/core/src/internal/io.js
@@ -163,7 +163,7 @@ export function select(selector = identity, ...args) {
 **/
 export function actionChannel(pattern, buffer) {
   if (process.env.NODE_ENV === 'development') {
-    check(pattern, is.notUndef, 'actionChannel(pattern,...): argument pattern is undefined')
+    check(pattern, is.pattern, 'actionChannel(pattern,...): argument pattern is not valid')
 
     if (arguments.length > 1) {
       check(buffer, is.notUndef, 'actionChannel(pattern, buffer): argument buffer is undefined')

